mlem/core/objects.py

```diff
@@ -668,6 +668,9 @@ def write_value(self) -> Artifacts:
     def load_value(self):
         self.dataset = self.reader.read(self.relative_artifacts)
 
+    def load_batch_value(self, batch: int):
```

Comment:
this should be changed to something like `read_batch`, return `Iterator[DatasetType]` and not set any values. `self.dataset` field is for whole dataset value, and if you wish to use batching, that means you don't want to load whole dataset into memory. So, `read_batch` will return lazy iterator that you can iterate on and get `DatasetType`s to work with. For now lets say that those `DatasetType`s should be the same as the `reader` is holding (==`DatasetType` of the whole dataset).